From 5eec1d9bbea59d2a80959e8443925e50bf82ee22 Mon Sep 17 00:00:00 2001
From: Jenny TC <jenny.tc@intel.com>
Date: Wed, 10 Dec 2014 15:49:28 +0530
Subject: [PATCH 1/1] bq24192: Fix charger online status on fault

When a charger fault triggers, the charger online should
be set to 0. The charger interrupts were not triggering since
they were masked. Unmasked the interrupt by changing the mask
bit in whiskey cove driver. Also checked the charger health
to indicate the online status

Change-Id: I235433bf2cc4c742309b7a6a4ff7ddbc0b53ed51
Signed-off-by: Jenny TC <jenny.tc@intel.com>
---
 drivers/mfd/intel_soc_pmic_wc.c | 2 +-
 drivers/power/bq24192_charger.c | 3 ++-
 2 files changed, 3 insertions(+), 2 deletions(-)

diff --git a/drivers/mfd/intel_soc_pmic_wc.c b/drivers/mfd/intel_soc_pmic_wc.c
index f012c7d..fba04e2 100644
--- a/drivers/mfd/intel_soc_pmic_wc.c
+++ b/drivers/mfd/intel_soc_pmic_wc.c
@@ -470,7 +470,7 @@ struct intel_pmic_irqregmap whiskey_cove_irqregmap[] = {
 		{THRM1IRQ, 7, 1, INTEL_PMIC_REG_W1C},
 	},
 	{ /* Ext. Chrgr */
-		{MIRQLVL1, 4, 0x1, 0},
+		{MCHGRIRQ, 0, 0x1, 0},
 		{CHGRIRQ, 0, 1, INTEL_PMIC_REG_W1C},
 		{CHGRIRQ, 0, 1, INTEL_PMIC_REG_W1C},
 	},
diff --git a/drivers/power/bq24192_charger.c b/drivers/power/bq24192_charger.c
index 27e0c8c..4f831c6 100644
--- a/drivers/power/bq24192_charger.c
+++ b/drivers/power/bq24192_charger.c
@@ -1509,7 +1509,8 @@ static int bq24192_usb_get_property(struct power_supply *psy,
 		}
 		break;
 	case POWER_SUPPLY_PROP_ENABLE_CHARGER:
-		val->intval = chip->is_charger_enabled;
+		val->intval = (bq24192_get_charger_health() ==
+				POWER_SUPPLY_HEALTH_GOOD);
 		break;
 	case POWER_SUPPLY_PROP_CHARGE_CONTROL_LIMIT:
 		val->intval = chip->cntl_state;
-- 
1.9.1

